OpenKeychain/src/main/java/org/sufficientlysecure/keychain/service/BackupNotificationService.java

```diff
@@ -0,0 +1,88 @@
+package org.sufficientlysecure.keychain.service;
+
+import android.app.AlarmManager;
+import android.app.NotificationManager;
+import android.app.PendingIntent;
+import android.app.Service;
+import android.content.Context;
+import android.content.Intent;
+import android.os.IBinder;
+import android.support.annotation.Nullable;
+
+import org.sufficientlysecure.keychain.receiver.BackupNotificationAlarmReceiver;
+import org.sufficientlysecure.keychain.util.Preferences;
+
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * This class is used to setup a backup reminder for every new key created by user
+ */
+public class BackupNotificationService extends Service {
+
+    public static final String ACTION_SET_REMINDER = "setReminder";
+    public static final String ACTION_DELETE_REMINDER = "deleteReminder";
+    public static final String ACTION_DELETE_ALL_REMINDERS = "deleteAllReminders" ;
+
+    public static final String MASTER_KEY_ID = "masterKeyId";
+    public static final String NAME = "name";
+    public static final String EMAIL = "email";
+
+    private void setBackupReminder(long mMasterKeyId, String name, String email) {
+        Preferences.getPreferences(this).setBackupReminder(mMasterKeyId);
+        Intent intentAlarm = new Intent(this, BackupNotificationAlarmReceiver.class);
+
+        intentAlarm.putExtra(BackupNotificationAlarmReceiver.MASTER_KEY_ID, mMasterKeyId);
+        intentAlarm.putExtra(BackupNotificationAlarmReceiver.NAME, name);
+        intentAlarm.putExtra(BackupNotificationAlarmReceiver.EMAIL, email);
+
+        PendingIntent pendingIntent = PendingIntent.getBroadcast(this,1,intentAlarm,
+                PendingIntent.FLAG_UPDATE_CURRENT);
+
+        AlarmManager alarmManager = (AlarmManager)this.getSystemService(Context.ALARM_SERVICE);
+        alarmManager.setInexactRepeating(AlarmManager.RTC_WAKEUP, System.currentTimeMillis(),
```

Comment:
I thought we will show notification after every 5 days till user does not create the backup. That's why it's repeating. But I see your point if he doesn't create the backup in the first time then we should not show notification again. He might not interested in creating backup at all. I will change it as you suggested. Thanks.